src/epics/helpers/ChatHandler.js

```diff
@@ -0,0 +1,208 @@
+import atob from 'atob';
+import btoa from 'btoa';
+import miniLock from '../../utils/miniLock';
+import { nowUTC } from '../../utils/time';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { ajax } from 'rxjs/observable/dom/ajax';
+import 'rxjs/add/operator/mergeMap';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mapTo';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/observable/from';
+import 'rxjs/add/observable/throw';
+
+class ChatHandler {
+
+  constructor(wsUrl) {
+    this.wsUrl = wsUrl;
+    this.wsMessageSubject = null;
+    this.wsUserStatusSubject = null;
+  }
+
+  onWsClose = (event) => {
+    console.error('Websocket got close event', event);
+    this.wsMessageSubject.error(new Error('The websocket connection was closed!'));
+  }
+
+  onWsError = (event) => {
+    console.error('Websocket got error event', event);
+    this.wsMessageSubject.error(new Error('Error occured on websocket connection!'));
+  }
+
+  onWsMessage = (event) => {
+    const data = JSON.parse(event.data);
+    if (data && data.ephemeral && data.ephemeral.length && data.ephemeral.length > 0) {
+      Observable.from(data.ephemeral)
+        .mergeMap(ephemeral =>
+          this.createDecryptEphemeralObservable({ ephemeral, mID: this.mID, secretKey: this.secretKey }))
+        .mergeMap(this.resolveIncomingMessageTypeObservable)
+        .catch(error => console.error('An error occured in ChatHandler', error))
```

Comment:
@Spetastian `occurred` :-) (with 2 `r`s)

Reply:
Pfft, try to learn swedish 😉

Reply:
Just kidding, will fix that 😄